pkg/disk/disk.go

```diff
@@ -58,6 +59,54 @@ const (
 	XBootLDRPartitionGUID = "BC13C2FF-59E6-4262-A352-B275FD6F7172"
 )
 
+// FSType is the filesystem type enum.
+//
+// There should always be one value for each filesystem type supported by
+// osbuild stages (stages/org.osbuild.mkfs.*) and the unset/none value.
+type FSType uint64
```

Comment:
What about making this a `string` alias, and use descriptive values? Should make debugging a tad bit simpler. Are there downsides (apart from slightly worse performance)?

Reply:
There's not much downside other than making it "easy" to use strings in place of the actual value sometimes, which can make it hard to change or track.  Having the enum backed by a uint says, a bit more clearly, "the underlying value doesn't matter", which I find encourages better use of the enum.

But I can go either way.

Reply:
Fwiw, I had the same thought and tried it out and it was the same about the amount of code/complexity, golang just does not have a good way of doing this it feels so  I did not comment (maybe I should have).

Reply:
Do we have any strong feelings either way?  Should I go with strings?

Reply:
I personally don't have strong feelings. :)